### PR TITLE
:bug: golangci-lint should use go-build tags appropriately

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: hack/boilerplate/boilerplate.py [hunk 1]
```diff
@@ -62,10 +62,6 @@ def get_refs():
     return refs
```
Comment (vincepri) [inline hack/boilerplate/boilerplate.py:62]: This file was updated from upstream https://github.com/kubernetes/kubernetes/blob/4dfd73940396730caf331e35cbb28235d233f2a0/hack/boilerplate/boilerplate.py

File: hack/tools/mdbook/tabulate/tabulate.go [hunk 2]
```diff
@@ -28,10 +29,13 @@ import (
 	"sigs.k8s.io/kubebuilder/docs/book/utils/plugin"
 )
 
-// Tabulate.
+// Tabulate adds support for switchable tabls in mdbook.
```
Comment (sbueringer) [inline hack/tools/mdbook/tabulate/tabulate.go:32]:
```suggestion
// Tabulate adds support for switchable tabs in mdbook.
```
nit

File: .golangci.yml [hunk 3]
```diff
@@ -149,6 +149,11 @@ issues:
     - stylecheck
     text: "ST1016: methods on the same type should have the same receiver name"
     path: .*(api|types)\/.*\/conversion.*\.go$
+  # hack/tools
+  - linters:
+    - typecheck
+    text: import (".+") is a program, not an importable package
+    path: ^tools\.go$
```
Comment (sbueringer) [inline .golangci.yml:156]: nit: should we match the entire path and not only the filename?

File: hack/tools/mdbook/embed/embed.go [hunk 4]
```diff
@@ -1,3 +1,4 @@
+//go:build tools
```
Comment (sbueringer) [inline hack/tools/mdbook/embed/embed.go:1]: Is there a way to enforce this going forward?

File: .golangci.yml (hunk 3 above)
Comment (vincepri) [reply .golangci.yml:156]: We can't, the path is relative to where golanci-lint runs, which is the tools folder in this case. It's not ideal but that's also why I added a stricter validation here

File: hack/tools/mdbook/embed/embed.go (hunk 4 above)
Comment (vincepri) [reply hack/tools/mdbook/embed/embed.go:1]: It's enforced by golangci-lint now, and you can use `make lint-fix`

File: hack/tools/mdbook/tabulate/tabulate.go (hunk 2 above)
Comment (vincepri) [reply hack/tools/mdbook/tabulate/tabulate.go:32]: tabls 😂

File: hack/boilerplate/boilerplate.py (hunk 1 above)
Comment (vincepri) [reply hack/boilerplate/boilerplate.py:62]: Added a readme, given that the whole folder was copied

File: .golangci.yml (hunk 3 above)
Comment (sbueringer) [reply .golangci.yml:156]: Oh yeah makes sense

**Comment by sbueringer** [comment]
Thx!
/lgtm

File: internal/envtest/webhooks.go [hunk 5]
```diff
@@ -67,7 +67,7 @@ func initWebhookInstallOptions() envtest.WebhookInstallOptions {
 	if err != nil {
 		klog.Fatalf("Failed to append bootstrap controller webhook config: %v", err)
 	}
-	controlplaneyamlFile, err := os.ReadFile(filepath.Join(root, "controlplane", "kubeadm", "config", "webhook", "manifests.yaml"))
+	controlplaneyamlFile, err := os.ReadFile(filepath.Join(root, "controlplane", "kubeadm", "config", "webhook", "manifests.yaml")) //nolint:Gosec
```
Comment (sbueringer) [inline internal/envtest/webhooks.go:70]:
```suggestion
	controlplaneyamlFile, err := os.ReadFile(filepath.Join(root, "controlplane", "kubeadm", "config", "webhook", "manifests.yaml")) //nolint:gosec
```
nit if you want

**Review by fabriziopandini — COMMENTED** [review]
More lint!
Thanks!
/approve
/lgtm

**Pull request merged**